pkgs/applications/networking/cluster/terraform/default.nix

```diff
@@ -37,6 +37,29 @@ let
         maintainers = with maintainers; [ jgeerds zimbatm peterhoeg ];
       };
     } // attrs');
+
+  terraform_0_10_2 = generic {
+    version = "0.10.2";
+    sha256 = "1q7za7jcfqv914a3ynfl7hrqbgwcahgm418kivjrac6p1q26w502";
+    patches = [ ./provider-path.patch ];
+    passthru = { inherit providers withPlugins; };
+  };
+
+  # TODO: make me stackable
+  withPlugins = plugins: stdenv.mkDerivation {
```

Comment:
this should also take `terraform:` as the first argument. Then set the passthru in the generic builder where `terraform` is a reference to self.